Priority Queue/FastPriorityQueue.cs

```diff
@@ -122,40 +122,49 @@ public void Enqueue(T node, float priority)
             _numNodes++;
             _nodes[_numNodes] = node;
             node.QueueIndex = _numNodes;
-            CascadeUp(_nodes[_numNodes]);
+            CascadeUp(node);
         }
 
         #if NET_VERSION_4_5
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         #endif
         private void Swap(T node1, T node2)
         {
+            // Cache to local for faster repeated access
+            int index1 = node1.QueueIndex;
+            int index2 = node2.QueueIndex;
+
             //Swap the nodes
-            _nodes[node1.QueueIndex] = node2;
-            _nodes[node2.QueueIndex] = node1;
+            _nodes[index1] = node2;
+            _nodes[index2] = node1;
 
             //Swap their indicies
-            int temp = node1.QueueIndex;
-            node1.QueueIndex = node2.QueueIndex;
-            node2.QueueIndex = temp;
+            node1.QueueIndex = index2;
+            node2.QueueIndex = index1;
```

Comment:
At the time I made this change, my test showed a 3% improvement in dequeue speed from this change alone. That said, I wouldn't be very surprised if that turned out to be random variation, and it seems I did indeed remove every call to it.